pkgs/development/python-modules/deal/default.nix

```diff
@@ -0,0 +1,107 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+
+  # build
+, flit-core
+, astroid
+
+  # tests
+, pytestCheckHook
+, docstring-parser
+, isort
+, marshmallow
+, pytest-cov
+, sphinx
+, hypothesis
+, vaa
+, deal-solver
+, pygments
+, typeguard
+, coverage
+, urllib3
+}:
+
+buildPythonPackage rec {
+  pname = "deal";
+  version = "4.23.3";
+  format = "pyproject";
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "life4";
+    repo = pname;
+    rev = "refs/tags/${version}";
+    hash = "sha256-duFxe2KSQQb7HB5KrrE32xzTb6QkQcrQssiuXLKao50=";
+  };
+
+  preCheck = ''
+    # don't do coverage
+    substituteInPlace pyproject.toml \
+      --replace "\"--cov-fail-under=100\"," "" \
+      --replace "\"--cov=deal\"," "" \
+      --replace "\"--cov-report=html\"," "" \
+      --replace "\"--cov-report=term-missing:skip-covered\"," ""
+  '';
+
+  nativeBuildInputs = [
+    flit-core
+  ];
+
+  propagatedBuildInputs = [
+    astroid
+    deal-solver
+    pygments
+    typeguard
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+
+    docstring-parser
+    isort
+    marshmallow
+    pytest-cov
+    sphinx
```

Comment:
Does this mean we are building docs we should install into the output?

Reply:
No. `deal` has a [sphinx integration](https://github.com/life4/deal/blob/master/docs/details/docs.md#sphinx-autodoc) which [gets checked](https://github.com/life4/deal/blob/a3b7ff4666424dca1a811812053f0ddea30e9889/tests/test_sphinx.py). Building the docs requires [a different command](https://github.com/life4/deal/blob/84832dfd30938a02abd3fe32ee834c86af1dc47d/Taskfile.yml#L104)